### PR TITLE
Add typehints to Mailer.

### PR DESCRIPTION
**Pull request by dereuromark**
Some doc blocks were incorrect as well.

Maybe we can also get rid of some |null things in favor of just string in the neat future.

### CODE REVIEW

**Comment by markstory** [comment]
There are a few relevant looking errors in the test results.

**Comment by markstory** [comment]
Refs #11935

File: src/Mailer/Email.php [hunk 1]
```diff
@@ -492,7 +492,7 @@ public function getReadReceipt()
      * @return $this
```
Comment (ADmad) [inline src/Mailer/Email.php]: Shouldn't we now change the docblock too to `@return self` to match actual typehint?

File: src/Mailer/Mailer.php [hunk 2]
```diff
@@ -212,7 +212,7 @@ public function viewBuilder()
      * @param array $args Method arguments
      * @return $this|mixed
      */
-    public function __call($method, $args)
+    public function __call(string $method, array $args)
```
Comment (ADmad) [inline src/Mailer/Mailer.php]: Adding typehints to magic methods seems quite unnecessary as these methods are called by PHP itself not the user.

Comment (dereuromark) [reply src/Mailer/Mailer.php]: OK

File: src/Mailer/Email.php (hunk 1 above)
Comment (dereuromark) [reply src/Mailer/Email.php]: I think we dont have to. one is php language, the other for IDE to understand chaining.

**Comment by dereuromark** [comment]
Found the issue.

    _smtpSend(?string $data, $checkCode = '250'): ?string

$checkCode is string|bool input
We should at some point fix this up to string|null input.
Not sure if that can be done for 4.0 now, though.

**Pull request merged**